_posts/2021/2021-09-16-5-most-valuable-things-i-learned-in-my-10-plus-year-career.md

```diff
@@ -0,0 +1,124 @@
+Many developers tend to overestimate the importance of their own domain in the system.
+
+Let's assume there are 3 views on that:
+- Business: How to be more successful as a company, and how can this be achieved?
+- User: How can we provide value to the end-user or customer with the code you write?
+- Technology: How to build software that is robust and lasts long (clean and homogeneous architecture, efficient, low memory usage etc.)?
+
+When you start working on new tasks, it is always helpful to know the answers to the what-why-how questions.
+Usually, these questions always helped me to get a grasp at the big picture:
+- have the end in mind (what?)
+- understand and challenge the requirements (why?)
+- connect the dots (how?)
+
+
+### 3. Become a T-Shaped person
+
+Many people argue if you should become a specialist or generalist in your industry.
+For me, the answer is a simple yes and no. But first, we need to understand what being a T-shaped person means:
+
+> A T-shaped person has deep knowledge/skills in one area and a broad base of general supporting knowledge/skills.
+
+I always had the ambition to have a broad knowledge.
+I wanted to know more than just the programming language and the framework I used to write my code.
+Additionally, I learned and tried to use new languages, concepts, new (web) technologies in simple pet projects.
+The opposite of a T-Shaped person is an I-shaped person who only has deep knowledge in one field and no general knowledge.
+There is also the dash-shaped person who only has general knowledge and no deep knowledge of a topic.
+
+But why should you aim for that goal?
+
+- Collaboration with other people is easier because you have at least some basic knowledge about other topics.
+- You stay interested in learning new things.
+- You can break the ice with new colleagues more easily.
+- As you become more and more familiar with concepts outside of your knowledge, you become slowly immune against the [Paradox of Expertise](https://www.psychologytoday.com/us/blog/tracking-wonder/201906/the-paradox-expertise){:target="_blank"}.
+- You will get tasks with more responsibility from your employer.
+
+
+
+### 4. Learn about software architecture (how to integrate external systems: file transfer, remote procedure, messaging)
+
+A lot of developers love to code.
+Design patterns will pop up along the developer journey for sure.
+Patterns like MVC, Strategy, Bridge, Adapter, Singleton are almost common knowledge for a developer.
+But there is a lot more to software architecture than this.
+For example, in modern software architecture, you have to integrate a lot of external services instead of implementing them by yourself.
+There are lots of ways to do this (e.g. Webservice based solutions, file-based integrations, and more can be found in this [book](https://www.amazon.de/Patterns-Enterprise-Application-Architecture-Addison-Wesley-ebook/dp/B008OHVDFM/ref=rvi_1/){:target="_blank"}).
```

Comment:
I would add "can be found, for example, in this book" - otherwise it sounds a bit like promo for the book, like we are presenting it as the single source of truth.